astropy/io/ascii/src/tokenizer.c

```diff
@@ -601,10 +601,15 @@ double str_to_double(tokenizer_t *self, char *str)
     {
         val = strtod(str, &tmp);
 
-        if (tmp == str || *tmp != 0)
+        if (tmp == str || *tmp != 0) {
+		    if (0 == strncmp(str, "nan", 3)) {
+				return NAN;
+		    }
```

Comment:
To be consistent with the Python reader (which in turn uses np.array to do type conversion), this should accept `inf` as well and be case insensitive (`InF` should work).